client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/InteractionSection/InteractionEventForm.tsx

```diff
@@ -0,0 +1,235 @@
+import {useFormContext, Controller } from 'react-hook-form';
+import {Grid, Typography, Divider, Collapse, Checkbox, FormControlLabel} from '@material-ui/core';
+
+import Contact from 'models/Contact';
+import Toggle from 'commons/Toggle/Toggle';
+import useFormStyles from 'styles/formStyles';
+import TimePick from 'commons/DatePick/TimePick';
+import FormInput from 'commons/FormInput/FormInput';
+import {get} from 'Utils/auxiliaryFunctions/auxiliaryFunctions';
+import placeTypesCodesHierarchy from 'Utils/placeTypesCodesHierarchy';
+import {getOptionsByPlaceAndSubplaceType} from 'Utils/placeTypesCodesHierarchy';
+import InteractionEventDialogData from 'models/Contexts/InteractionEventDialogData';
+import PlacesTypesAndSubTypes, {PlacesTypesAndSubTypesProps} from 'commons/Forms/PlacesTypesAndSubTypes/PlacesTypesAndSubTypes';
+import InteractionEventDialogFields from 'models/enums/InteractionsEventDialogContext/InteractionEventDialogFields';
+
+import AddressForm from './AddressForm/AddressForm';
+import useStyles from './InteractionEventFormStyles';
+import PlaceNameForm from './PlaceNameForm/PlaceNameForm';
+import BusinessContactForm from './BusinessContactForm/BusinessContactForm';
+
+export const defaultContact: Contact = {
+    firstName: '',
+    lastName: '',
+    phoneNumber: '',
+    idNumber: '',
+    contactType: -1,
+    creationTime: new Date(),
+    involvedContact: null,
+};
+
+const InteractionEventForm: React.FC<InteractionEventFormProps> = (
+    { onPlaceSubTypeChange, isVisible, interactionData, isNewInteraction }: InteractionEventFormProps): JSX.Element => {
+    const {control, watch, clearErrors, setValue, errors, setError} = useFormContext();
+
+    const placeType = watch(InteractionEventDialogFields.PLACE_TYPE);
+    const placeSubType = watch(InteractionEventDialogFields.PLACE_SUB_TYPE);
+    const interactionStartTime = watch(InteractionEventDialogFields.START_TIME);
+    const interactionEndTime = watch(InteractionEventDialogFields.END_TIME);
+    const isUnknownTime = watch(InteractionEventDialogFields.UNKNOWN_TIME);
+    const locationAddress = watch(InteractionEventDialogFields.LOCATION_ADDRESS);
+    const placeName = watch(InteractionEventDialogFields.PLACE_NAME);
+    const placeDescription = watch(InteractionEventDialogFields.PLACE_DESCRIPTION);
+
+    const [startTime, setStartTime] = useState<Date | null>(isNewInteraction ? null : interactionStartTime);
+    const [endTime, setEndTime] = useState<Date | null>(isNewInteraction ? null : interactionEndTime);
+
+    const formConfig = useMemo(() => getOptionsByPlaceAndSubplaceType(placeType, placeSubType), [placeType, placeSubType]);
+
+    const classes = useStyles();
+    const formClasses = useFormStyles();
+
+    const handleTimeChange = (currentTime: Date, interactionDate: Date, fieldName: string) => {
+        if (isValid(currentTime)) {
+            let newDate = new Date(interactionDate.getTime());
```

Comment:
I think newDate can be a const, in addition delete space under

Reply:
same as below